spring-cloud-stream/src/main/java/org/springframework/cloud/stream/converter/CompositeMessageConverterFactory.java

```diff
@@ -42,16 +42,16 @@
 
 	private final ObjectMapper objectMapper;
 
-	private final List<AbstractFromMessageConverter> converters;
+	private final List<MessageConverter> converters;
 
 	public CompositeMessageConverterFactory() {
-		this(Collections.<AbstractFromMessageConverter>emptyList(), new ObjectMapper());
+		this(Collections.<MessageConverter>emptyList(), new ObjectMapper());
 	}
 
 	/**
-	 * @param customConverters a list of {@link AbstractFromMessageConverter}
+	 * @param customConverters a list of {@link AbstractMessageConverter}
 	 */
-	public CompositeMessageConverterFactory(List<? extends AbstractFromMessageConverter> customConverters,
+	public CompositeMessageConverterFactory(List<? extends MessageConverter> customConverters,
```

Comment:
why not keep `AbstractMessageType` as the type within the list considering anything else would be ignored based on line 91 (why permit anything else to be added?)